tikv-client/src/main/java/com/pingcap/tikv/types/EnumType.java

```diff
@@ -92,9 +92,19 @@ private Integer parseEnumValue(Integer number) throws ConvertOverflowException {
   /** {@inheritDoc} */
   @Override
   protected Object decodeNotNull(int flag, CodecDataInput cdi) {
-    if (flag != Codec.UVARINT_FLAG)
-      throw new TypeException("Invalid EnumType(IntegerType) flag: " + flag);
-    int idx = (int) IntegerCodec.readUVarLong(cdi) - 1;
+    int idx;
+    switch (flag) {
+        // encodeKey write UINT_FLAG
```

Comment:
`encodeKey` -> `decodeKey`

Reply:
What I want to comment here is that UINT_FLAG is used during encodeKey, so we need to handle it in decode

Reply:
sounds good